src/main/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaData.java

```diff
@@ -134,6 +137,9 @@ public int getPrecision(int column) {
         return 1;
       case Types.DATE:
         return 10;
+      case Types.REAL:
+        return JdbcDataType.FLOAT32.getPrecision();
+      case Types.FLOAT:
       case Types.DOUBLE:
         return 14;
```

Comment:
Is there a source for these numbers? They don't seem correct to me.

IIUC, scale is the number of digits to the right of the decimal point, and precision is the total number of digits. So precision should be at least scale+1? 

The current numbers for DOUBLE (precision: 14, scale: 15), and the new numbers for float (7 and 7) don't seem right. 
At least for FLOAT, I would expect scale to be 7, precision 8 (or scale 6, precision 7)

Reply:
The precision for `FLOAT32` is based on this: https://download.oracle.com/otn_hosted_doc/jdeveloper/904preview/jdk14doc/docs/guide/jdbc/getstart/mapping.html (search for REAL for information on float32 in JDBC). Precision 7 is also what SQL Server and MySQL return for float32.
PostgreSQL returns precision 8 for `real` / `float4`, but with the following comment: 

```
        // For float4 and float8, we can normally only get 6 and 15
        // significant digits out, but extra_float_digits may raise
        // that number by up to two digits.
```

What to return for `scale` for floating point numbers in JDBC seems to be a bit hard to determine. In this case, I've followed PostgreSQL (partly because we also support the PG dialect), which returns `scale==precision` for floating point numbers. See https://github.com/pgjdbc/pgjdbc/blob/818953a2b30480ecc874f812e46fcb82d1a395fd/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L812 and https://github.com/pgjdbc/pgjdbc/blob/818953a2b30480ecc874f812e46fcb82d1a395fd/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L865

MySQL and SQL Server return 0 as the scale for floating point numbers, most probably based on the comment in the JDBC documentation that `getScale(..)` should '0 is returned for data types where the scale is not applicable.' (see https://docs.oracle.com/javase/8/docs/api/java/sql/ResultSetMetaData.html#getScale-int-)

Reply:
Thanks for the pointers! Current changes LGTM with this context